src/bench/readblock.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+#include <bench/data.h>
+
+#include <consensus/validation.h>
+#include <node/blockstorage.h>
+#include <streams.h>
+#include <test/util/setup_common.h>
+#include <util/chaintype.h>
+#include <validation.h>
+
+static FlatFilePos WriteBlockToDisk(ChainstateManager& chainman) {
+    CDataStream stream{benchmark::data::block413567, SER_NETWORK, PROTOCOL_VERSION};
+    CBlock block;
+    stream >> block;
+
+    return chainman.m_blockman.SaveBlockToDisk(block, 0, nullptr);
+}
+
+static void ReadBlockFromDiskTest(benchmark::Bench& bench)
+{
+    const auto testing_setup{MakeNoLogFileContext<const TestingSetup>(ChainType::MAIN)};
+    ChainstateManager& chainman{*testing_setup->m_node.chainman};
+
+    CBlock block;
+    const auto pos{WriteBlockToDisk(chainman)};
+
+    bench.run([&] {
+        assert(chainman.m_blockman.ReadBlockFromDisk(block, pos));
```

Comment:
as per developer notes, [assertions shouldn't have side-effects](https://github.com/bitcoin/bitcoin/blob/624333455a5745a7f184d0df531dc348d0ac48dd/doc/developer-notes.md?plain=1#L742), so the following would be preferred:
```suggestion
        bool success{chainman.m_blockman.ReadBlockFromDisk(block, pos)};
        assert(success);
```
(same for the second test below)

Reply:
Agree that my suggestion is non-blocking, this is obviously much less critical for benchmarks and tests as opposed to production code. Having it enforced on the CI would be very nice indeed (I might look into how well this works with clang-tidy and if it'd also catch our own assertion-variants), I personally wouldn't remove the section from the docs though.

Reply:
Done